native/dlapmt.go

```diff
@@ -0,0 +1,88 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import "github.com/gonum/blas/blas64"
+
+// Dlapmt rearranges the columns of the m×n matrix x as specified by the
+// permutation k[0], k[1], ..., k[n-1] of the integers 0, ..., n-1.
+//
+// If forward is true a forward permutation is performed:
+//
+//  x[*, k[j]] is moved x[*, j] for j = 0, 1, ..., n-1.
+//
+// otherwise a  backward permutation is performed:
+//
+//  x[*, j] is moved to x[*, k[j]] for j = 0, 1, ..., n-1.
+//
+// k must have length n, otherwise Dlapmt will panic otherwise. k is zero-indexed.
+//
+// Dlapmt is an internal routine. It is exported for testing purposes.
+func (impl Implementation) Dlapmt(forward bool, m, n int, x []float64, ldx int, k []int) {
+	checkMatrix(m, n, x, ldx)
+	if len(k) != n {
+		panic(badKperm)
+	}
+
+	if n <= 1 {
+		return
+	}
+
+	for i, v := range k {
+		v++
+		k[i] = -v
+	}
+
+	bi := blas64.Implementation()
+
+	if forward {
+		for i, v := range k {
+			if v > 0 {
+				continue
+			}
+			j := i + 1
+			in := -v
+			k[j-1] = in
+			for {
+				if k[in-1] > 0 {
+					break
+				}
+
+				// for ii := 0; ii < m; ii++ {
```

Comment:
These commented-out section are just to show what I have replaced with the `Dswap` call. I can leave them in or take them out.